src/mumble/Overlay_win.cpp

```diff
@@ -49,52 +49,72 @@ typedef void (__cdecl *PrepProc)();
 extern "C" __declspec(dllexport) void mumbleSelfDetection() {};
 
 OverlayPrivateWin::OverlayPrivateWin(QObject *p) : OverlayPrivate(p) {
-	hpInstall = NULL;
-	hpRemove = NULL;
-	qlOverlay = new QLibrary(this);
-
-	QString path = QString::fromLatin1("%1/mumble_ol.dll").arg(qApp->applicationDirPath());
-	qlOverlay->setFileName(path);
-	if (! qlOverlay->load()) {
-		QMessageBox::critical(NULL, QLatin1String("Mumble"), tr("Failed to load overlay library. This means either that:\n"
-		                      "- the library (mumble_ol.dll) wasn't found in the directory you ran Mumble from\n"
-		                      "- you're on an OS earlier than WinXP SP2"), QMessageBox::Ok, QMessageBox::NoButton);
-		qWarning("Overlay failure");
-		return;
+	m_helper_exe_path = QString::fromLatin1("%1/mumble_ol.exe").arg(qApp->applicationDirPath());
+	m_helper_exe_args = QStringList(QString::number(OVERLAY_MAGIC_NUMBER));
+	m_helper_process = new QProcess(this);
+
+	connect(m_helper_process, SIGNAL(started()),
+	        this, SLOT(helperProcessStarted()));
+
+	connect(m_helper_process, SIGNAL(finished(int, QProcess::ExitStatus)),
+	        this, SLOT(helperProcessExited(int, QProcess::ExitStatus)));
+}
+
+OverlayPrivateWin::~OverlayPrivateWin() {
+	m_active = false;
+}
+
+void OverlayPrivateWin::startHelper(QProcess *helper) {
+	if (helper->state() == QProcess::NotRunning) {
+		helper->start(m_helper_exe_path, m_helper_exe_args);
+	} else {
+		qWarning("OverlayPrivateWin: startHelper() called while process is already running. skipping.");
 	}
+}
 
-	GetOverlayMagicVersionProc gompvp = (GetOverlayMagicVersionProc)qlOverlay->resolve("GetOverlayMagicVersion");
-	if (! gompvp) {
-		qWarning("The overlay librarys overlay protocol version could not be verified. Overlay will not be enabled.");
-		return;
+void OverlayPrivateWin::setActive(bool active) {
+	if (m_active != active) {
+		m_active = active;
+
+		if (m_active) {
+			startHelper(m_helper_process);
+		} else {
+			m_helper_process->terminate();
```

Comment:
The whole thing is wrapped in `if (m_active != active) { ... }`, so we only take action if the value actually changes.